src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4202,65 +4202,74 @@ private BoundExpression BindImplicitObjectCreationExpression(ImplicitObjectCreat
 
         protected BoundExpression BindObjectCreationExpression(ObjectCreationExpressionSyntax node, BindingDiagnosticBag diagnostics)
         {
-            var typeWithAnnotations = BindType(node.Type, diagnostics);
-            var type = typeWithAnnotations.Type;
-            var originalType = type;
+            BoundExpression result = bindObjectCreationExpression(node, diagnostics);
 
-            if (typeWithAnnotations.NullableAnnotation.IsAnnotated() && !type.IsNullableType())
-            {
-                diagnostics.Add(ErrorCode.ERR_AnnotationDisallowedInObjectCreation, node.Location, type);
-            }
+            Debug.Assert(result is not BoundBadExpression { ChildBoundNodes: var children } || !children.Any((child, node) => child.Syntax == node, node));
```

Comment:
>result is not BoundBadExpression { ChildBoundNodes: var children } || !children.Any((child, node) => child.Syntax == node, node) [](http://example.com/codeflow?start=25&length=128)

Since we're using this assertion twice, should we perhaps make it a conditional method that more explicitily elaborates what the condition violation breaks?

Reply:
> Since we're using this assertion twice, should we perhaps make it a conditional method that more explicitily elaborates what the condition violation breaks?

Any name suggestions for the method? I decided to avoid this problem all together. Didn't feel like there is much code duplication to avoid.

Reply:
`AssertNoDuplicatedBadExpressionSyntax`?

Reply:
It is not really about syntax though. The syntax is not the root cause of the problem, it is not what is causing the problem. The syntax check is a good enough simplification  to detect a possible problem. The Assert doesn't have to be 100% accurate as long as it doesn't produce false violations.